contracts/stellar-interchain-token-service/src/token_manager.rs

```diff
@@ -26,13 +26,8 @@ impl TokenManagerClientExt for TokenManagerClient<'_> {
     fn mint(&self, env: &Env, token_address: &Address, recipient: &Address, amount: i128) {
         let _: Val = self.execute(
             token_address,
-            &Symbol::new(env, "mint_from"),
-            &vec![
-                env,
-                self.address.to_val(),
-                recipient.to_val(),
-                amount.into_val(env),
-            ],
+            &Symbol::new(env, "mint"),
+            &vec![env, recipient.to_val(), amount.into_val(env)],
```

Comment:
add a separate `mint_from` and `mint`. the latter is used for the new mint burn type, while the former is preserved for native interchain tokens